### PR TITLE
Improve CI

### LINKED ISSUE #199: Verify component ids match public keys in CI
We can use [this code](https://github.com/brave/brave-core-crx-packager/blob/master/lib/util.js#L106C1-L114C2) for it. Should help to protect against something like #198 happening again.

### PR DESCRIPTION
**Pull request by antonok-edm**
closes #199

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "1.0.0",
   "description": "Custom resources and scriptlets used for Brave's adblocker",
   "main": "index.js",
+  "type": "module",
```
Comment (ShivanKaul) [inline package.json:6]: why?

File: .github/workflows/json_verify.yml [hunk 2]
```diff
@@ -1,30 +0,0 @@
-name: CI
```
Comment (ShivanKaul) [inline .github/workflows/json_verify.yml:1]: why?

File: verify.js [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
 
-const assert = require('node:assert')
-const test = require('node:test')
-const { Engine, FilterFormat, FilterSet } = require('adblock-rs')
+import assert from 'node:assert'
+import crypto from 'crypto'
+import test from 'node:test'
+import { Engine, FilterFormat, FilterSet } from 'adblock-rs'
+
+const getIDFromBase64PublicKey = (key) => {
```
Comment (ShivanKaul) [inline verify.js:8]: I thought you said we already have this code somewhere? Can we not just export that function?

File: .github/workflows/json_verify.yml (hunk 2 above)
Comment (antonok-edm) [reply .github/workflows/json_verify.yml:1]: It's still there; just moved it to be combined into the other workflow

File: package.json (hunk 1 above)
Comment (antonok-edm) [reply package.json:6]: required for `import` and stuff

File: verify.js (hunk 3 above)
Comment (antonok-edm) [reply verify.js:8]: It's deep in `brave-core-crx-packager` internals. I considered using it directly but it doesn't seem worth it to expose that and pull it in as a new dependency here just for CI purposes

**Review by ShivanKaul — APPROVED** [review]
LGTM, @bcaller can you take a look

**Pull request merged**